casper/src/main/scala/coop/rchain/casper/api/BlockAPI.scala

```diff
@@ -77,6 +78,12 @@ object BlockAPI {
     ).raiseError[F, ApiErr[String]]
     val readOnlyCheck = readOnlyError.whenA(isNodeReadOnly)
 
+    // Check if deploy's shardId equals to node shardId
+    val shardIdError = new RuntimeException(
+      "Deploy was rejected because its shard is not equal node shard"
```

Comment:
Please change the error message to include more info what it's expected and what is seen.

Something like
`Deploy shardId '${d.data.shardId}' is not as expected network shard '$shardId'.`

Reply:
Yes, I agree. Resolved in [commit](https://github.com/rchain/rchain/pull/3631/commits/2e69f99787d9e3143f44082b9c4e0bfe2f9382bb).